### PR TITLE
Improved duel code to bring in additional units

### LINKED ISSUE #776: LS infest and duels
if ls infests a creep pre duel then he and his creep get spawned in the floor of the duel arena

### PR DESCRIPTION
**Pull request by Chronophylos**
Completing #524
Fixed #776 by casting Lifestealer Control when inside a creep

Further testing is needed

### CODE REVIEW

**Comment by Haganeko** [comment]
"valve pls"

**Comment by Chronophylos** [comment]
Fixing LS infest too

File: game/scripts/vscripts/components/duels/duels.lua [hunk 1]
```diff
@@ -453,7 +450,38 @@ function Duels:AllPlayers (state, cb)
...
+                                     nil,
+                                     FIND_UNITS_EVERYWHERE,
+                                     DOTA_UNIT_TARGET_TEAM_FRIENDLY,
+                                     DOTA_UNIT_TARGET_BASIC,
+                                     DOTA_UNIT_TARGET_FLAG_PLAYER_CONTROLLED,
+                                     FIND_ANY_ORDER,
+                                     false)
+  for _,child in pairs(children) do
+    if child:HasMovementCapability() then
+      self:SafeTeleport(child, location, maxDistance)
```
Comment (chrisinajar) [inline game/scripts/vscripts/components/duels/duels.lua]: Don't you need to confirm that unit belongs to that hero, or you'll end up with all of the team's units on top of whoever was teleported in last?

Comment (Chronophylos) [inline game/scripts/vscripts/components/duels/duels.lua]: maybe

**Comment by Haganeko** [comment]
Can LS fix be applied to other hiding abilities?

**Comment by Chronophylos** [comment]
what do you mean?

**Comment by Chronophylos** [comment]
like Astral Imprisonment?

**Comment by Haganeko** [comment]
Every ability which hides the hero seems to cause issues with the duel teleportation. I was wondering if whatever you did to fix LS could fix that.

**Comment by Chronophylos** [comment]
I made LS cast Control to fix it.

**Comment by Haganeko** [comment]
Oh.

**Comment by Chronophylos** [comment]
I'm testing if I can just remove the modifier

**Comment by Chronophylos** [comment]
works

**Pull request merged**